lib/unit-ruby/util/resource_operations.rb

```diff
@@ -77,5 +77,25 @@ def save(attributes)
         update_resource_from_json_api(updated_resource)
       end
     end
+
+    # Replace the entire resource, instead of only patching dirty attributes
+    module Replace
+      def replace(attributes)
+        updated_resource = self.class.connection.put(
+          self.class.resource_path(id),
+          {
+            data: {
+              type: resource_type,
+              attributes: attributes
+            }
+          }
+        )
+        update_resource_from_json_api(updated_resource)
+      end
+
+      def replace_json(json_attributes)
```

Comment:
I slightly preferred having a separate method instead of a `json: true` argument for `#replace` because it is a bit more clear what it's intended for.

Reply:
Good call, and this approach - even though it has the extra step of parsing then re-serializing - also gives us validation that we're sending valid json to the API.